### PR TITLE
AS-128: BMI Collection, plus some refactoring and framework for stats and stat endpoints.

### PR DESCRIPTION
**Pull request by joberstein**
Added BMI Collection functionality. Adjusted the containers to use state instead of props for changing properties, updated the api file to reduce code duplication, added a framework for the stat and stats endpoints.

### CODE REVIEW

File: src/actions/student.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
-export const STUDENT_FETCH_REQUESTED = 'STUDENT_FETCH_REQUESTED';
-export const STUDENT_FETCH_SUCCEDED = 'STUDENT_FETCH_SUCCEDED';
-export const STUDENT_FETCH_FAILED = 'STUDENT_FETCH_FAILED';
+export const STUDENT_FETCH_REQUESTED = 'STATS_FETCH_REQUESTED';
+export const STUDENT_FETCH_SUCCEEDED = 'STATS_FETCH_SUCCEEDED';
+export const STUDENT_FETCH_FAILED = 'STATS_FETCH_FAILED';
```
Comment (brianeroth) [inline src/actions/student.js]: Should these be STUDENT_FETCH_REQUESTED instead of STATS_FETCH_REQUESTED?

Comment (cooper627) [reply src/actions/student.js]: I called it that for the searching of a student and adding of a student and not for getting the stats of a student that is already within a program.

Comment (cooper627) [reply src/actions/student.js]: Probably shouldve been in a AddStudent action file instead of the students file

Comment (joberstein) [reply src/actions/student.js]: Hmm not sure how that happened, I'll fix that.

**Pull request merged**